### PR TITLE
Add Nickname Support for Thanking, and migrated thanksListener to Gui…

### PR DESCRIPTION
**Pull request by helloimalastair**
# Pull Request

## Description

Changes the ThanksListener, to allow support for nicknames when sending a thanks response. Migrated from `message.mentions.users` to `message.mentions.members`.

## Documentation

For _any_ version updates, please verify if the [documentation page](https://www.beccalyria.com/discord-documentation) needs an update. If it does, please [create an issue there](https://github.com/BeccaLyria/discord-documentation/issues/new?assignees=nhcarrigan&labels=%F0%9F%9A%A6+status%3A+awaiting+triage&template=update.md&title=%5BUPDATE%5D) to ensure it is not forgotten.

- [x] My contribution does NOT require a documentation update.
- [ ] My contribution DOES require a documentation update.

### CODE REVIEW

File: src/listeners/thanksListener.ts [hunk 1]
```diff
@@ -18,28 +18,28 @@ export const thanksListener: ListenerInt = {
 
       const thanksRegex =
         /((?:^|\s)(?:(?:th(?:n[qx]|x)|t[xyq]|tn(?:[x]){0,2})|\w*\s*[.,]*\s*than[kx](?:[sxz]){0,2}|than[kx](?:[sxz]){0,2}(?:[uq]|y(?:ou)?)?)|grazie|arigato(?:[u]{0,1})|doumo|gracias?|spasibo|dhanyavaad(?:hamulu)?|o?brigad(?:o|a)|dziekuje|(?:re)?merci|multumesc|shukra?an|danke)\b/gi;
-      if (!thanksRegex.test(content) || !mentions.users.size) {
+      if (!thanksRegex.test(content) || !mentions.members.size) {
```
Comment (nhcarrigan) [inline src/listeners/thanksListener.ts:21]:
```suggestion
      if (!thanksRegex.test(content) || !mentions.members?.size) {
```

Unlike `users`, `members` can be undefined (as is the case in DMs, which Becca doesn't support) - using optional chaining here will invoke the early return and resolve the build error.

**Review by nhcarrigan — CHANGES_REQUESTED** [review]

**Review by nhcarrigan — APPROVED** [review]
This is looking good to me. @helloimalastair did you have more you wanted to do? Or is this good to merge?

**Comment by helloimalastair** [comment]
> 
> 
> This is looking good to me. @helloimalastair did you have more you wanted to do? Or is this good to merge?

Nothing that I can think of, but I try to keep different changes in different pull requests, so if I come up with something, I'll add another one.

**Comment by nhcarrigan** [comment]
## Congratulations

Congratulations on making your first pull request to our project. We have reviewed your contributions and are happy to accept them.

We hope you enjoyed the contribution process, and encourage you to watch for issues labelled `help wanted` for additional opportunities to contribute.

**Pull request merged**